LandSandBoat/server
# replace aH categories in item_basic.sql with vars for readability

**_I affirm:_**

## What does this pull request do?
Replaces numerical aH categories in item_basic.sql with SQL variables for ease of readability and maintenance.

## Steps to test these changes
Since this is a massive change, and I'll admit I'm a bit scared, below is the python script i wrote to make the changes.
You should be able to execute this from the same folder that item_basic.sql resides in to generate the updated file.

```python
# Define the mapping of values to variables
value_map = {
    "0": "@NONE",
    "1": "@H2H",
    "2": "@DAGGER",
    "3": "@SWORD",
    "4": "@GREATSWORD",
    "5": "@AXE",
    "6": "@GREATAXE",
    "7": "@SCYTHE",
    "8": "@POLEARM",
    "9": "@KATANA",
    "10": "@GREATKATANA",
    "11": "@CLUB",
    "12": "@STAFF",
    "13": "@BOW",
    "14": "@INSTRUMENTS",
    "15": "@AMMUNITION",
    "16": "@SHIELD",
    "17": "@HEAD",
    "18": "@BODY",
    "19": "@HANDS",
    "20": "@LEGS",
    "21": "@FEET",
    "22": "@NECK",
    "23": "@WAIST",
    "24": "@EARRINGS",
    "25": "@RINGS",
    "26": "@BACK",
    "27": "@UNUSED",
    "28": "@WHITE_MAGIC",
    "29": "@BLACK_MAGIC",
    "30": "@SUMMONING",
    "31": "@NINJUTSU",
    "32": "@SONGS",
    "33": "@MEDICINES",
    "34": "@FURNISHINGS",
    "35": "@CRYSTALS",
    "36": "@CARDS",
    "37": "@CURSED_ITEMS",
    "38": "@SMITHING",
    "39": "@GOLDSMITHING",
    "40": "@CLOTHCRAFT",
    "41": "@LEATHERCRAFT",
    "42": "@BONECRAFT",
    "43": "@WOODWORKING",
    "44": "@ALCHEMY",
    "45": "@GEOMANCER",
    "46": "@MISC",
    "47": "@FISHING_GEAR",
    "48": "@PET_ITEMS",
    "49": "@NINJA_TOOLS",
    "50": "@BEAST_MADE",
    "51": "@FISH",
    "52": "@MEAT_EGGS",
    "53": "@SEAFOOD",
    "54": "@VEGETABLES",
    "55": "@SOUPS",
    "56": "@BREADS_RICE",
    "57": "@SWEETS",
    "58": "@DRINKS",
    "59": "@INGREDIENTS",
    "60": "@DICE",
    "61": "@AUTOMATON",
    "62": "@GRIPS",
    "63": "@ALCHEMY_2",
    "64": "@MISC_2",
    "65": "@MISC_3",
    "255": "@INVALID"
}

# Open the item_basic.sql file for reading
with open("item_basic.sql", "r") as file:
    # Read all lines from the file
    sql_lines = file.readlines()

# Process each line
for line in sql_lines:
    # Check if the line starts with "INSERT INTO"
    if line.startswith("INSERT INTO"):
        # Split the line by commas
        values = line.split(',')
        # Get the value in the 7th column
        value_7th_column = values[6].strip()
        # Replace the value with the corresponding variable
        if value_7th_column in value_map:
            values[6] = value_map[value_7th_column]
        # Reconstruct the line
        new_line = ','.join(values)
        # Print the modified line
        print(new_line, end='')
    else:
        # Print unchanged lines
        print(line, end='')
```
